src/main/java/com/artipie/asto/memory/BenchmarkStorage.java

```diff
@@ -65,8 +81,34 @@ public BenchmarkStorage(final InMemoryStorage backend) {
     }
 
     @Override
-    public CompletableFuture<Collection<Key>> list(final Key prefix) {
-        throw new NotImplementedError("Not implemented yet");
+    public CompletableFuture<Collection<Key>> list(final Key root) {
+        return CompletableFuture.supplyAsync(
+            () -> {
+                final String prefix = root.string();
+                final Collection<Key> keys = new TreeSet<>(Key.CMPRTR_STRING);
+                final SortedSet<String> bckndkeys = this.backend.data
+                    .navigableKeySet()
+                    .tailSet(prefix);
+                final SortedSet<Key> lclkeys = this.local
+                    .navigableKeySet()
+                    .tailSet(new Key.From(prefix));
+                for (final String keystr : bckndkeys) {
+                    if (keystr.startsWith(prefix) && !this.deleted.contains(new Key.From(keystr))) {
+                        keys.add(new Key.From(keystr));
+                    } else {
+                        break;
+                    }
+                }
+                for (final Key key : lclkeys) {
+                    if (key.string().startsWith(prefix) && !this.deleted.contains(key)) {
+                        keys.add(key);
+                    } else {
+                        break;
+                    }
+                }
```

Comment:
Why to implement it manually? You can just use: `addAll` and `removeAll`:
```
set = local.keySet()
set.addAll(backend.keySet()
set.removeAll(deleted)
```

Reply:
@g4s8 there is a test case in `StorageListTest` which would fail in case of using `addAll`
```
...
final byte[] data = "some data!".getBytes();
final BlockingStorage blocking = new BlockingStorage(storage);
blocking.save(new Key.From("1"), data);
blocking.save(new Key.From("a", "b", "c", "1"), data);
blocking.save(new Key.From("a", "b", "2"), data);
blocking.save(new Key.From("a", "z"), data);
blocking.save(new Key.From("z"), data);
final Collection<String> keys = blocking.list(new Key.From("a", "b"))
...
```
It happens because `tailset` returns also keys which are greater than prefix (e.g. `z` and `a/z` in this case)